metadata.yaml

```diff
@@ -7,21 +7,25 @@ description: |
 
 containers:
   udm:
-    resource: udm-image
+    resource: sdcore-udm-image
```

Comment:
Why add the `sdcore` prefix? We've used `<nf>-image` in all other charms

Reply:
I've taken the pattern from @Gmerold [UDR](https://github.com/canonical/sdcore-udr-operator/blob/43e104901edd4d4c7eb2ec14001b3dc4702e7770/tests/integration/test_integration.py#L32), it's for fetching the image in the integration tests like so:
```python
resources = {
        f"{APPLICATION_NAME}-image": METADATA["resources"][f"{APPLICATION_NAME}-image"][
            "upstream-source"
        ],
    }
```
I like the pattern, and also I think we should strive for consistency. I'd change it in the rest of the charms whenever we mess with the integration tests.

Reply:
I'm not sure about associating the resource image with the application. This necessarily assumes 1 workload per app and won't work with the upf for instance.

Reply:
Reverted.